workflow/controller/dag.go

```diff
@@ -344,7 +345,7 @@ func (woc *wfOperationCtx) executeDAGTask(dagCtx *dagContext, taskName string) {
 
 		if node.Completed() {
 			// Run the node's onExit node, if any.
-			hasOnExitNode, onExitNode, err := woc.runOnExitNode(task.OnExit, task.Name, node.Name, dagCtx.boundaryID, dagCtx.tmplCtx)
+			hasOnExitNode, onExitNode, err := woc.runOnExitNode(ctx, task.OnExit, task.Name, node.Name, dagCtx.boundaryID, dagCtx.tmplCtx)
```

Comment:
Hmmm... in this function we add a `ctx` to a bunch of signatures just for this method. Makes me wonder if we should just save the context in `woc.ctx`? That way we don't have to pass it in a huge stack of functions just for one or two uses. @alexec What do you think?

Reply:
AFAIK you should rarely have `Context` as a field, it is intended to be passed around as the first parameter to functions. But I get that is is a bit onerous.

Reply:
I'm down with whatever y'all want to go with, but my suggestion would be to keep it as the first argument in functions because it's more idiomatic: https://blog.golang.org/context